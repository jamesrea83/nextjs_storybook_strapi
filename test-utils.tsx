import { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { Themes } from './styles/themes';

interface Props {
	children: ReactNode;
}

const Wrapper = ({ children }: Props) => (
	<ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
	render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
