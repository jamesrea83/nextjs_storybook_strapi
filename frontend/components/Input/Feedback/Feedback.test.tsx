import { render } from '@/test-utils';

import { Feedback } from '.';

describe('Feedback tests', () => {
	it('should render valid result', () => {
		const { asFragment } = render(<Feedback isValid>Good</Feedback>);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should render invalid result', () => {
		const { asFragment } = render(<Feedback isValid={false}>Bad</Feedback>);

		expect(asFragment()).toMatchSnapshot();
	});
});
