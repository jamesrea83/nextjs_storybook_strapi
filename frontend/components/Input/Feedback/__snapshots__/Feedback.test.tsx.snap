// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback tests should render invalid result 1`] = `
<DocumentFragment>
  <span
    class="css-8k225a"
  >
    Bad
  </span>
</DocumentFragment>
`;

exports[`Feedback tests should render valid result 1`] = `
<DocumentFragment>
  <span
    class="css-1wbdfck"
  >
    Good
  </span>
</DocumentFragment>
`;
